src/lib/update-lock.ts

```diff
@@ -0,0 +1,104 @@
+import * as Bluebird from 'bluebird';
+import * as lockFileLib from 'lockfile';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as path from 'path';
+import * as Lock from 'rwlock';
+
+import constants = require('./constants');
+import { ENOENT, UpdatesLockedError } from './errors';
+
+type asyncLockFile = typeof lockFileLib & {
+	unlockAsync(path: string): Bluebird<void>;
+	lockAsync(path: string): Bluebird<void>;
+};
+const lockFile = Bluebird.promisifyAll(lockFileLib) as asyncLockFile;
+export type LockCallback = (
+	appId: number,
+	opts: { force: boolean },
+	fn: () => PromiseLike<void>,
+) => Bluebird<void>;
+
+function baseLockPath(appId: number): string {
+	return path.join('/tmp/balena-supervisor/services', appId.toString());
+}
+
+export function lockPath(appId: number, serviceName: string): string {
+	return path.join(baseLockPath(appId), serviceName);
+}
+
+function lockFilesOnHost(appId: number, serviceName: string): string[] {
+	return ['updates.lock', 'resin-updates.lock'].map(filename =>
+		path.join(constants.rootMountPoint, lockPath(appId, serviceName), filename),
+	);
+}
+
+const locksTaken: { [lockName: string]: boolean } = {};
+
+// Try to clean up any existing locks when the program exits
+process.on('exit', () => {
+	const result = [];
+	for (const lockName of _.keys(locksTaken)) {
+		try {
+			result.push(lockFile.unlockSync(lockName));
+			// tslint:disable-next-line
```

Comment:
What?

Reply:
We want to ignore the error, but we can't add an empty catch block without the linter complaining. Any better ideas?

Reply:
Oh nice, yeah that works.